fr/integrations/data-analytics/sc-elastic-events.md

```diff
@@ -1,35 +1,268 @@
+
+All those parameters are documented **[here](https://github.com/centreon/centreon-stream-connector-scripts/blob/master/modules/docs/sc_param.md#default-parameters)**.
+
+Some of them are overridden by this stream connector.
+
+| Type   | Name                | Default value for the stream connector |
+| ------ | ------------------- | -------------------------------------- |
+| string | accepted_categories | neb                                    |
+| string | accepted_elements   | host_status,service_status             |
+
+## Event bulking
+
+This stream connector is compatible with event bulking. Meaning that it is able to send more that one event in each call to the Elastic REST API.
+
+To use this feature you must add the following parameter in your stream connector configuration.
+
+| Type   | Name            | Value           |
+| ------ | --------------- | --------------- |
+| number | max_buffer_size | `more than one` |
+
+## Event format
+
+This stream connector will send event with the following format.
+
+### service_status event
+
+```json
+{
+    "event_type": "host",
+    "status": "CRITICAL",
+    "state": "2",
+    "state_type": 1,
+    "host": "my_host",
+    "service": "a_service",
+    "output": "CRITICAL: Burnin and Lootin",
+    "timestamp": 1637229207
+}
+```
+
+### host_status event
+
+```json
+{
+    "event_type": "host",
+    "status": "DOWN",
+    "state": "1",
+    "state_type": 1,
+    "host": "my_host",
+    "output": "CRITICAL: No woman no cry",
+    "timestamp": 1637229207
+}
+```
+
+### Custom event format
+
+This stream connector allows you to change the format of the event to suit your needs. Only the **event** part of the json is customisable. It also allows you to handle events type that are not handled by default such as **ba_status events**.
+
+In order to use this feature you need to configure a json event format file and add a new stream connector parameter.
+
+| Type   | Name        | Value                                           |
+| ------ | ----------- | ----------------------------------------------- |
+| string | format_file | /etc/centreon-broker/elastic-events-format.json |
+
+> The event format configuration file must be readable by the centreon-broker user
+
+To learn more about custom event format and templating file, head over the following **[documentation](https://github.com/centreon/centreon-stream-connector-scripts/blob/master/modules/docs/templating.md#templating-documentation)**.
+
+## Curl commands
+
+Here is the list of all the curl commands that are used by the stream connector.
+
+### Send events
+
+```shell
+curl -u elastic:centreon-es-passwd --header 'content-type: application/json'  -X POST "<elastic_url>/_bulk" --data-binary '{"index":{"_index":"<elastic_index_status>"}}\n{"host":"jamaica","satus":"OK","state_type":1,"state":0,"timestamp":<a_recent_timestamp>,"event_type":"service","service":"kingston","output":"OK: Everything is gonna be alright"}\n'
```

Comment:
I couldn't make it works using \n in the curl. I had to do it the old school way using the enter key from my keyboard :D 

+ there is a typo in the "satus" instead of "status"

```suggestion
curl -u elastic:centreon-es-passwd --header 'content-type: application/json'  -X POST "<elastic_url>/_bulk" --data-binary '{"index":{"_index":"<elastic_index_status>"}}
{"host":"jamaica","status":"OK","state_type":1,"state":0,"timestamp":<a_recent_timestamp>,"event_type":"service","service":"kingston","output":"OK: Everything is gonna be alright"}
'
```